### ISSUE TITLE
Shelly Plus i4 is not discovered

### ISSUE THREAD

**Issue by AbsoluteBlack0** [bieniu/ha-shellies-discovery-gen2#220]
**Describe the bug**
I have a Shelly Dimmer 2 and a Shelly Plus i4 in use. A freshly installed HomeAssistant on a Pi4. Everything updated.

Dimmer2 as a Gen1 device has been found without any problems. The i4 as Gen2 device is not found.And to be honest I'm clueless why...

**Expected behavior**
Device and Entities discovered and working.

**Versions:**
 - Home Assistant:   2023.6.0
Supervisor 2023.06.1
Operating System 10.2
Frontend 20230607.0 
 - Shellies Discovery GEN2: 2.6.1
 - Shelly device firmware: 1.0.0beta3

**Shellies Discovery GEN2 automation:**
![2023-06-09 01_56_34-Studio Code Server – Home Assistant](https://github.com/bieniu/ha-shellies-discovery-gen2/assets/105823270/27983f3b-7b9b-4202-ba08-1fb47ce37c8d)

![2023-06-09 01_56_48-Studio Code Server – Home Assistant](https://github.com/bieniu/ha-shellies-discovery-gen2/assets/105823270/a0670e43-4d44-43f8-9849-021d2a479b53)

![2023-06-09 01_55_52-Shelly PlusI4](https://github.com/bieniu/ha-shellies-discovery-gen2/assets/105823270/3491438b-77db-493a-9b94-602b2812a0c0)

**Debug log:**
[home-assistant_2023-06-09T00-20-03.589Z.log](https://github.com/bieniu/ha-shellies-discovery-gen2/files/11697438/home-assistant_2023-06-09T00-20-03.589Z.log)

As you can see Gen1 device discovery is working:

![2023-06-09 02_21_45-Settings – Home Assistant](https://github.com/bieniu/ha-shellies-discovery-gen2/assets/105823270/8b75507d-1ec4-4543-9611-905a169a0d54)

**Comment by bieniu** [comment]
More information needed. Please attach a full log.

**Comment by AbsoluteBlack0** [comment]
I'm new to everything. Mosquitto is pretty much a standard installation as of now. Customize is not active and I did not write an ACL file. I can try that and give my user unrestricted permissions on all topics so the device will have permission.

The log in my first post is the full log HomeAssistant created. There was nothing more.

While playing around today I found something out.
First I gave my i4 a custom prefix "shellies/" and added that prefix in the automation.
After a while I started the Announce Gen2 Automation manually. And suddenly:


![2023-06-10 00_21_21-Settings – Home Assistant](https://github.com/bieniu/ha-shellies-discovery-gen2/assets/105823270/1998e899-b4d8-4953-9322-5776e3f048cd)


Diagnostics of that device:

[mqtt-bd3795c57b4ee158584fe4e009de7f9b-Shelly Plus I4-95c9ad6a0e68bc9bf2af9810c2b556c2.json.txt](https://github.com/bieniu/ha-shellies-discovery-gen2/files/11710453/mqtt-bd3795c57b4ee158584fe4e009de7f9b-Shelly.Plus.I4-95c9ad6a0e68bc9bf2af9810c2b556c2.json.txt)


I'm still missing all Sensors. Especcially for the Buttons. They are configured as Pushbuttons not Switches.
